appnav/src/main/kotlin/io/element/android/appnav/LoggedInFlowNode.kt

```diff
@@ -368,32 +363,22 @@ class LoggedInFlowNode @AssistedInject constructor(
         }
     }
 
-    suspend fun attachRoomList() {
-        if (!canShowRoomList()) return
-        attachChild<Node> {
-            backstack.singleTop(NavTarget.RoomList)
-        }
-    }
-
     suspend fun attachRoom(roomId: RoomId) {
-        if (!canShowRoomList()) return
+        waitForNavTargetAttached { navTarget ->
+            navTarget is NavTarget.RoomList
+        }
         attachChild<RoomFlowNode> {
-            backstack.singleTop(NavTarget.RoomList)
             backstack.push(NavTarget.Room(roomId.toRoomIdOrAlias()))
         }
     }
 
-    private fun canShowRoomList(): Boolean {
-        return ftueService.state.value is FtueState.Complete
```

Comment:
By the way, after a cold boot, `ftueService.state.value` was equal to `Unknown` which was the root cause of the issue this PR is fixing.